src/Servers/HttpSys/src/RequestProcessing/RequestContext.cs

```diff
@@ -326,6 +326,10 @@ private void SetFatalResponse(int status)
 
         internal unsafe void Delegate(DelegationRule destination)
         {
+            if (destination == null)
```

Comment:
Nullability should do a better job of validating this. Could we remove this check given that it's a non-public API?

Reply:
@pranavkm HttpSys.FeatureContext.DelegateRequest which is public calls this method directly. I could move this check over there but figured it made more sense here since this is where it's used. 

I don't quite get your point about nullability.